x/src/cargo.rs

```diff
@@ -86,39 +107,65 @@ impl Cargo {
         self
     }
 
+    // Passes extra environment variables to be passed to the executable.
+    pub fn env_additions<I, S>(&mut self, envs: I) -> &mut Self
+    where
+        I: IntoIterator<Item = (S, S)>,
+        S: AsRef<OsStr>,
+        //        S: std::convert::AsRef<std::ffi::OsStr>,
+    {
+        self.inner.envs(envs);
+        self
+    }
+
     pub fn run(&mut self) -> Result<()> {
```

Comment:
Is `inner` in a valid state at the end of this execution? If it isn't, then this should cause the `Cargo` to be consumed (i.e. `pub fn run(self)`).

Reply:
Interesting.   I'll try consuming the Cargo instance and see what happens....  inner can have its status called multiple times, so appears to be valid during the function body after it's execution....

Reply:
Yeah it might be in a state where you can access it to get some more details, but maybe not in one where `run` can be called again. That would be a perfect use case for consuming `self`.

Reply:
Good point, This API was supposed to mimic the API of Command (and the builder pattern as it is makes it nicer to have `run` not take self) but given some of the things that are happening under the hood (pass through args and the like) it doesn't quite workout. Though this could be easily fixed by 1 of 2 options: 1) take `self` as you described (but that does limit the ability to rerun the command if we want to later) or 2) don't actually construct a `Command` till the run function and instead hold on to all the args till `run` is called